src/main/java/net/fabricmc/fabric/api/client/itemgroup/FabricItemGroup.java

```diff
@@ -0,0 +1,35 @@
+package net.fabricmc.fabric.api.client.itemgroup;
+
+import net.fabricmc.fabric.client.itemgroup.ItemGroupExtensions;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Identifier;
+
+import java.util.function.Supplier;
+
+public final class FabricItemGroup {
+
+	Identifier identifier;
+	Supplier<ItemStack> stackSupplier = () -> ItemStack.EMPTY;
+
+	public FabricItemGroup(Identifier identifier) {
+		this.identifier = identifier;
+	}
+
+	public FabricItemGroup icon(Supplier<ItemStack> stackSupplier){
+		this.stackSupplier = stackSupplier;
+		return this;
+	}
+
+	public ItemGroup create(){
```

Comment:
This is a builder, just call it FabricItemGroupBuilder, add a create(Identifier) and rename this to build if so.